### DISCUSSION TITLE
opening an album fails

### DISCUSSION THREAD

**Discussion by jelinj8** [rern/rAudio#106] [Fixed Issues]
I had a bit tweaed rAudio instance before, so I've tried a clean install of rAudio (just configured network + timezone), connected an USB flash with two albums.
The scan was completed, one album works, but an attempt to open the other in library fails. This is what is in journalctl:

```
Jan 13 13:26:43 rAudio-Kuba nginx[330]: 2021/01/13 13:26:43 [error] 330#330: *20 FastCGI sent in stderr: "PHP message: PHP Notice:  Only variables should be passed by reference in /srv/http/mpdlibrary.php on line 145PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Un
```

The UI hangs with a flashing logo.

Album (the problematic one) uploaded on GDrive: https://drive.google.com/drive/folders/1WmhW25HNosK5e4eLkiZUS2VilR4Zi8SI?usp=sharing

**Comment by rern** [comment]
It's encoding issue. Try converting both filenames and metadata tags to UTF-8.

**Reply by jelinj8** [reply]
Trouble is that they both are UTF-8 (or should be), valid ID3v1 + v2.3. I've just copied them from an UTF-8 based Ubuntu, maybe the filesystem on the flashdisk is a problem. But they are scanned correctly by MPD and shown in the GUI with correct accents.
I'm waiting for a new complete rebuild of library, but it might correlate with the empty albums list I've reported in other thread.
![image](https://user-images.githubusercontent.com/3816495/104465503-63192a80-55b4-11eb-9cc3-e462cae9d574.png)

**Comment by jelinj8** [comment]
it fails even when I try to open it from the Files view:
`2021/01/13 15:57:47 [error] 337#337: *185 FastCGI sent in stderr: "PHP message: PHP Notice:  Only variables should be passed by reference in /srv/http/mpdlibrary.php on line 145PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 406PHP message: PHP Notice:  Undefined variable: cue in /srv/http/mpdlibrary.php on line 403PHP message: PHP Warning:  A non-numeric value encountered in /srv/http/mpdlibrary.php on line 404PHP message: PHP Notice:  U`

I'm going to investigate a bit in the code (/srv/http/mpdlibrary.php:145).

**Comment by jelinj8** [comment]
Fine, I've probably found the problem:
`$ext = end( explode( '.', $plfiles[ 0 ] ) );`
The folder contains an m3u playlist, that is (quite legally) encoded in ANSI-1250.
I'll try to remove it (or change it to UTF-8 m3u8), but this would be quite common case.

**Comment by jelinj8** [comment]
Fine, confirming that it is the m3u file. It doesn't contain any metadata, just filenames, but they have accents encoded in ANSI-1250.
This might be quite an issue as lots of albums will have similar issue (be it 1250, 1252 or any other local codepage). MPD takes that OK, but the GUI chokes on that.

**Comment by rern** [comment]
Fixed.
